telethon_examples/payment.py

```diff
@@ -0,0 +1,174 @@
+                query_id=event.query_id,
+                success=True,
+                error=None
+            )
+        )
+    else:
+        # for example, something went wrong (whatever reason). We can tell customer about that:
+        await bot(
+            functions.messages.SetBotPrecheckoutResultsRequest(
+                query_id=event.query_id,
+                success=False,
+                error="Something went wrong"
+            )
+        )
+
+    raise events.StopPropagation
+
+
+# That event is handled at the end, when customer payed.
+@bot.on(events.Raw(types.UpdateNewMessage))
+async def payment_received_handler(event):
+    if isinstance(event.message.action, types.MessageActionPaymentSentMe):
+        payment: types.MessageActionPaymentSentMe = event.message.action
+        # do something after payment was recieved
+        if payment.payload.decode("UTF-8") == 'product A':
+            await bot.send_message(event.message.from_id, "Thank you for buying product A!")
+        elif payment.payload.decode("UTF-8") == "product B":
+            await bot.send_message(event.message.from_id, "Thank you for buying product B!")
+        raise events.StopPropagation
+
+
+# let's put it in one function for more easier way
+def generate_invoice(price_label: str, price_amount: int, currency: str, title: str,
+                     description: str, payload: str, start_param: str) -> types.InputMediaInvoice:
+    price = types.LabeledPrice(label=price_label, amount=price_amount)  # label - just a text, amount=10000 means 100.00
+    invoice = types.Invoice(
+        currency=currency,  # currency like USD
+        prices=[price],  # there could be a couple of prices.
+        test=True,  # if you're working with test token
+
+        #  params for requesting specific fields
+        name_requested=False,
+        phone_requested=False,
+        email_requested=False,
+        shipping_address_requested=False,
+
+        #  if price changes depending on shipping
+        flexible=False,
+
+        #  send data to provider
+        phone_to_provider=False,
+        email_to_provider=False
+
+    )
+    return types.InputMediaInvoice(
+        title=title,
+        description=description,
+        invoice=invoice,
+        payload=payload.encode("UTF-8"),  # payload, which will be sent to next 2 handlers
+        provider=provider_token,
+
+        provider_data=types.DataJSON("{}"),
+        # data about the invoice, which will be shared with the payment provider. A detailed description of
+        # required fields should be provided by the payment provider.
+
+        start_param=start_param,
+        # start_param will be passed with UpdateBotPrecheckoutQuery,
+        # I don't really know why is it needed, I guess like payload.
+        # from PTB docs: Unique deep-linking parameter that can be used to
+        #                 generate this invoice when used as a start parameter.

```

Comment:
We should probably reword this and there is no need to mention "PTB" (I'm sure a lot of people won't know what it is, nor should they care).

Something like:
```
# deep linking may also be used in UpdateBotPrecheckoutQuery
# see: https://core.telegram.org/bots#deep-linking
```

If it can be left empty, also note `# it may be the empty string if not needed`.